src/components/deploymentConfig/DeploymentConfig.tsx

```diff
@@ -641,20 +720,52 @@ export default function DeploymentConfig({
 
     const toggleSaveChangesModal = () => {
         dispatch({ type: DeploymentConfigStateActionTypes.toggleSaveChangesModal })
+        setSaveChangesCheckbox(!saveChangesCheckbox)
+
+    }
+
+    const closeConfirmationDialog = () => {
+        dispatch({
+            type: DeploymentConfigStateActionTypes.showConfirmation,
+            payload: false,
+        })
+        setSaveChangesCheckbox(!saveChangesCheckbox)
+
+    }
+    const handleChangeCheckbox = () => {
+        if (!saveChangesCheckbox) {
+            setSaveChangesCheckbox(!saveChangesCheckbox)
+            handleSubmit()
+        } else {
+            state.showSaveChangsModal && toggleSaveChangesModal()
+            state.showConfirmation && closeConfirmationDialog()
+        }
     }
 
     const toggleDraftComments = () => {
         dispatch({ type: DeploymentConfigStateActionTypes.toggleDraftComments })
     }
 
-    const prepareDataToSave = (skipReadmeAndSchema?: boolean) => {
+    const prepareDataToSave = (skipReadmeAndSchema?: boolean, saveEligibleChanges?: boolean) => {
+        let valuesOverride = obj
+        if(state.showLockedTemplateDiff) {
+            // if locked keys 
+            // @TODO: pass correct params
+            if(!lockedKeysArray.allowed) {
+                valuesOverride = YAML.parse(YAML.stringify(getUnlockedJSON(lockedOverride, lockedKeysArray.config)))
+            } else {
+                // if allowed keys
+                valuesOverride = YAML.parse(YAML.stringify(getLockedJSON(lockedOverride, lockedKeysArray.config)))
+            }
```

Comment:
remove YAML.stringify if not needed

Reply:
removed the YAML parsing need to give JSON to payload